offline/single_hetzner_machine_installation.md

```diff
@@ -15,440 +15,61 @@ If not using Hetzner, for reference, the specs of the ax101 server are:
-
-```
-apt update
-apt upgrade -y
-```
-
-### Install tmate
-
-Tmate is a terminal sharing service, which you might need in order for more than one person to collaborate on solving issues, Wire might ask you for a tmate session when debugging any problem you encounter.
-
-```
-sudo apt install tmate
-```
-
-If asked, to start a tmate session, you would simply then do:
-
-```
-tmate
-```
-
-And copy/paste the links that are generated, which would then result in the terminal session being shared with whomever you shared the links with.
-
-### Reboot
-
-reboot to load a new, patched kernel.
-
-```
-reboot
-```
-
-### Disable password login for sshd
-
-Install `nano` or your favorite text editor:
-
-```
-sudo apt install nano -y
-```
-
-Make sure the following values are configured in /etc/ssh/sshd_config:
-
-```
-# this is the important value
-PasswordAuthentication no
-
-# make sure PAM and Challenge Response is also disabled
-ChallengeResponseAuthentication no
-UsePAM no
-
-# don't allow root to login via password
-PermitRootLogin prohibit-password
-```
-
-Alternatively you can use these commands:
-
-```
-sed -i 's/#PasswordAuthentication yes/PasswordAuthentication no/g' /etc/ssh/sshd_config
-sed -i 's/UsePAM yes/ChallengeResponseAuthentication no\nUsePAM no/g' /etc/ssh/sshd_config
-sed -i 's/PermitRootLogin yes/PermitRootLogin prohibit-password/g' /etc/ssh/sshd_config
-```
-
-### re-start SSH
-
-```
-service ssh restart
-```
-
-### Install fail2ban
-
-```
-apt install fail2ban -y
-```
-
-## Create demo user.
-
-### create our 'demo' user
-
-```
-adduser --disabled-password --gecos "" demo
-```
-
-### copy ssh key to demo user
-
-```
-mkdir ~demo/.ssh
-cp ~/.ssh/authorized_keys /home/demo/.ssh/
-chown demo.demo ~demo/.ssh/
-chown demo.demo ~demo/.ssh/authorized_keys
-```
-
-### add a configuration for demo not to need a password in order to SUDO.
-
-```
-echo "demo ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/10-demo_user
-chmod 440 /etc/sudoers.d/10-demo_user
-```
-
-## ssh in as demo user.
-
-on the remote machine:
-
-```
-logout
-```
-
-on the local machine:
-
-```
-ssh -i ~/.ssh/id_ed25519 demo@65.21.197.76 -o serveraliveinterval=60
-```
-
-## Disable root login via ssh
-
-Use sudo to edit `/etc/ssh/sshd_config`
-
-```
-sudo nano /etc/ssh/sshd_config
-```
-
-And set the following:
-
-```
-# even better: don't allow to login as root via ssh at all
-PermitRootLogin no
-```
-
-Or use this command:
-
-```
-sudo sed -i 's/PermitRootLogin prohibit-password/PermitRootLogin no/g' /etc/ssh/sshd_config
-```
-
-### re-start SSH
+## Adjust playbook vars as needed
 
+Take a look at the "vars:" section in wire-server-deploy/ansible/hetzner-single-deploy.yml and adjust vars as needed. Example:
 ```
-sudo service ssh restart
+  vars:
+    artifact_hash: a6e0929c9a5f4af09655c9433bb56a4858ec7574
+    ubuntu_version: 22.04.3
+    ssh_pubkey: "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIODDzgw4BncNvSVOIraAXZYkkLn+iTo6ixkXXQ4lKZhL jan.schumacher@wire.com"
```

Comment:
might want to just change the key comment here and consider not using a valid key. i am sure a user might accidentally copy this one day :)